src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -5446,5 +5447,3416 @@ interface I14
+//                                               to expose this information? 
+@"
+IPropertyReferenceOperation: System.Int32 I1.P01 { get; set; } (Static) (OperationKind.PropertyReference, Type: System.Int32) (Syntax: 'T.P01')
+  Instance Receiver: 
+    null
+");
+
+            var m02 = compilation1.GetMember<MethodSymbol>("Test.M02");
+
+            Assert.Equal("System.Int32 I1.P01 { get; set; }", ((CSharpSemanticModel)model).LookupSymbols(node.SpanStart, m02.TypeParameters[0], "P01").Single().ToTestDisplayString());
+        }
+
+        [Fact]
+        public void ConsumeAbstractStaticPropertySet_03()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static int P01 { get; set; }
+}
+
+class Test
+{
+    static void M02<T>() where T : I1
+    {
+        T.P01 = 1;
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var verifier = CompileAndVerify(compilation1, verify: Verification.Skipped).VerifyDiagnostics();
+
+            verifier.VerifyIL("Test.M02<T>()",
+@"
+{
+  // Code size       15 (0xf)
+  .maxstack  1
+  IL_0000:  nop
+  IL_0001:  ldc.i4.1
+  IL_0002:  constrained. ""T""
+  IL_0008:  call       ""void I1.P01.set""
+  IL_000d:  nop
+  IL_000e:  ret
+}
+");
+
+            var tree = compilation1.SyntaxTrees.Single();
+            var model = compilation1.GetSemanticModel(tree);
+            var node = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().First().Left;
+
+            Assert.Equal("T.P01", node.ToString());
+            VerifyOperationTreeForNode(compilation1, model, node,
+// PROTOTYPE(StaticAbstractMembersInInterfaces): It feels like the "T" qualifier is important for this invocation, but it is not 
+//                                               reflected in the IOperation tree. Should we change the shape of the tree in order
+//                                               to expose this information? 
+@"
+IPropertyReferenceOperation: System.Int32 I1.P01 { get; set; } (Static) (OperationKind.PropertyReference, Type: System.Int32) (Syntax: 'T.P01')
+  Instance Receiver: 
+    null
+");
+
+            var m02 = compilation1.GetMember<MethodSymbol>("Test.M02");
+
+            Assert.Equal("System.Int32 I1.P01 { get; set; }", ((CSharpSemanticModel)model).LookupSymbols(node.SpanStart, m02.TypeParameters[0], "P01").Single().ToTestDisplayString());
+        }
+
+        [Fact]
+        public void ConsumeAbstractStaticPropertyCompound_03()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static int P01 { get; set; }
+}
+
+class Test
+{
+    static void M02<T>() where T : I1
+    {
+        T.P01 += 1;
+    }
+
+    static string M03<T>() where T : I1
+    {
+        return nameof(T.P01);
```

Comment:
I didn't understand why we also test `nameof` here.

Reply:
> I didn't understand why we also test nameof here.

We should be able to successfully reference the property inside a `nameof`.

---
In reply to: [613622102](https://github.com/dotnet/roslyn/pull/52605#discussion_r613622102) [](ancestors = 613622102)